### PR TITLE
Add Python factorial script

### PR DESCRIPTION
**Pull request by andrey-unblocked**
python/factorial.py — iterative factorial with CLI entry.

### CODE REVIEW

**Comment by andrey-unblocked** [comment]
[TESTING] No tests for `factorial`. Boundary cases (0, 1, negative → `ValueError`, large n) are exactly the spots that quietly break on refactor. A 5-line `test_factorial.py` with `pytest` parametrize would lock the contract.

File: python/factorial.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""Iterative factorial."""
+from __future__ import annotations
+import sys
+
+
+def factorial(n: int) -> int:
```
Comment (andrey-unblocked) [inline python/factorial.py:6]: No tests cover this function. Boundary cases (`0`, `1`, negative → `ValueError`, large `n`) are the spots that quietly regress on refactor — a short `test_factorial.py` with `pytest.mark.parametrize` would lock the contract.

Comment (andrey-unblocked) [inline python/factorial.py:6]: No tests cover this function. Boundary cases (0, 1, negative → ValueError, large n) are the spots that quietly regress on refactor — a short test_factorial.py with pytest.mark.parametrize would lock the contract.
